File: app/src/main/java/com/joshrap/liteweight/repositories/workouts/requests/CreateWorkoutRequest.java
package com.joshrap.liteweight.repositories.workouts.requests;

import com.joshrap.liteweight.models.workout.Routine;
import com.joshrap.liteweight.repositories.BodyRequest;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = false)
public class CreateWorkoutRequest extends BodyRequest {

    private String name;
    private SetRoutineRequest routine;
    private boolean setAsCurrentWorkout;

    public CreateWorkoutRequest(String name, Routine routine, boolean setAsCurrentWorkout) {
        this.name = name;
        this.routine = new SetRoutineRequest(routine);
        this.setAsCurrentWorkout = setAsCurrentWorkout;
    }
}

File: api/LiteWeightApi/Api/Workouts/Requests/CreateWorkoutRequest.cs
using System.ComponentModel.DataAnnotations;
using LiteWeightAPI.Imports;

namespace LiteWeightAPI.Api.Workouts.Requests;

public class CreateWorkoutRequest
{
	/// <summary>
	/// Name of the workout.
	/// </summary>
	/// <example>High Intensity Workout</example>
	[Required]
	[MaxLength(Globals.MaxWorkoutNameLength)]
	public string Name { get; set; } = null!;

	/// <summary>
	/// Routine of the workout.
	/// </summary>
	[Required]
	public SetRoutineRequest Routine { get; set; } = null!;

	/// <summary>
	/// If true, set this new workout to be the current workout for the user creating the workout.
	/// </summary>
	public bool SetAsCurrentWorkout { get; set; } = true;
}